Sally7/Internal/TypeExtensions.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System.Runtime.CompilerServices;
+
+namespace Sally7.Internal;
+
+internal static class TypeExtensions
+{
+    public static ref byte GetOffset(this ref byte destination, int offset)
+    {
+        return ref Unsafe.Add(ref destination, offset);
```

Comment:
Thanks, I think you actually made a similar comment last week on AdsClient, I'll try to keep this in mind!

Reply:
Also, since offset will always be positive anyway, as will all the lengths returned from `WireFormatting` methods, I could just as well move to `uint` in all those places I guess.

Reply:
Yep, `uint` works too.
As said in the AdsClient, it depends on where the cast to `uint` happens. Once here in this method, or when getting the `uint`-casted length of the span (which is `int`). I prefer having only one cast.
But you'll see when you make the change.

Reply:
Bummer, this is net6, 7 and 8 only and based on conversion to `UIntPtr` / `nuint`. I wasn't even targeting net6 yet. So now I'm wondering whether I want to go `uint` all around and cast it to `int` on < net6 or the other way around.

Reply:
Ah, then something like `Unsafe.Add(ref src, (IntPtr)(uint)idx);` works when the language given mapping `UIntPtr <-> nuint` isn't available (and the implicit cast from `uint -> nuint`).
So I'd keep the parameters as `int` and do the casts in that method.